src/clr-angular/data/datagrid/datagrid-row.ts

```diff
@@ -214,14 +198,41 @@ export class ClrDatagridRow implements AfterContentInit {
         });
 
         // Used to set things up the first time but only after all the columns are ready.
-        this.subscription = this.hideableColumnService.columnListChange.subscribe((columnList) => {
+        this.subscriptions.push(this.hideableColumnService.columnListChange.subscribe((columnList) => {
             // Prevents cell updates when cols and cells array are not aligned - only seems to run on init / first time.
             if (columnList.length === this.dgCells.length) {
                 this.updateCellsForColumns(columnList);
             }
+        }));
+        this.dgCells.filter((cell, index) => index > 0).forEach((cell) => {
+            this.scrollableCells.insert(cell.view);
         });
     }
 
+    ngAfterViewInit() {
+        this.subscriptions.push(this.displayMode.view.subscribe(viewChange => {
+            // Listen for view changes and move cells around depending on the current displayType
+            while (this.scrollableCells.detach()) {
+            }  // remove cells containers
+            while (this.calculatedCells.detach()) {
+            }  // remove cells containers
+            if (viewChange === DatagridDisplayMode.CALCULATE) {
+                this.dgCells.forEach(cell => {
+                    this.calculatedCells.insert(cell.view);
+                });
+                this.displayCells = false;
+            } else {
+                this.dgCells.forEach(cell => {
+                    this.scrollableCells.insert(cell.view);
+                });
+                this.displayCells = true;
+            }
+        }));
+        // Insert views the first time? Fixed the pagination issue after discussing w/ Eude.
+        this.dgCells.forEach(cell => this.scrollableCells.insert(cell.view));
```

Comment:
If the displaymode service was a behaviorsubject and sent the default mode as display, you wouldn't need this extra default thing here.

Reply:
BehaviorSubject ftw.

Reply:
I was able to remove similar code in `datagrid.ts`.